clients/wallet/src/error.rs

```diff
@@ -31,6 +31,36 @@ pub enum Error {
 }
 
 impl Error {
+	pub fn is_recoverable(&self) -> bool {
+		match self {
+			Error::HttpFetchingError(e) if e.is_timeout() => true,
+			Error::HorizonSubmissionError { title: _, status, reason: _, envelope_xdr: _ }
+				if *status == 504 =>
+				true,
+			Error::CacheError(e) => match e.kind {
+				CacheErrorKind::CreateDirectoryFailed |
+				CacheErrorKind::FileCreationFailed |
+				CacheErrorKind::WriteToFileFailed |
+				CacheErrorKind::DeleteFileFailed |
+				CacheErrorKind::FileDoesNotExist => true,
+				_ => false,
+			},
+			_ => false,
+		}
+	}
+
+	pub fn is_server_error(&self) -> bool {
+		match self {
+			Error::HttpFetchingError(e)
+				if e.status() == Some(reqwest::StatusCode::INTERNAL_SERVER_ERROR) =>
+				true,
+			Error::HorizonSubmissionError { title: _, status, reason: _, envelope_xdr: _ }
+				if *status == 500 =>
+				true,
```

Comment:
Should we add a range to this so that all 5xx errors are declared as server errors? Probably not that relevant but per definition of HTTP all 5xx errors would be server errors IIRC.

Reply:
done.